.github/workflows/reusable-docker-build.yml

```diff
@@ -0,0 +1,142 @@
+name: docker
```

Comment:
What's going to be the strategy to maintain this workflow? Its mostly the same one we have in our private actions repo and it really should stay mostly the same; How can we help make sure of that? Could we have a script that copies it from the private repo to here and use it as is (or maybe with some slight modifications to the upstream version so its compatible with this?)